packages/truffle-contract/lib/utils.js

```diff
@@ -117,49 +141,77 @@ var Utils = {
   },
 
   // Extracts optional tx params from a list of fn arguments
-  getTxParams: function(args){
+  getTxParams: function(methodABI, args) {
     var constructor = this;
 
-    var tx_params =  {};
+    var expected_arg_count = methodABI ? methodABI.inputs.length : 0;
+
+    tx_params = {};
     var last_arg = args[args.length - 1];
 
-    // It's only tx_params if it's an object and not a BigNumber.
-    if (Utils.is_object(last_arg) && !Utils.is_big_number(last_arg)) {
+    if (
+      args.length === expected_arg_count + 1 &&
+      Utils.is_tx_params(last_arg)
```

Comment:
The way this was done before caused an off-by-one if the last constructor arg was an `object` that wasn't a `BigNumber`